src/lib/lib.rs

```diff
@@ -43,6 +107,76 @@ pub async fn download_file(client: &reqwest::Client, url: &str, path: &str) -> R
     Ok(())
 }
 
+/// Clones ReShade shaders from their repositories
+pub fn clone_reshade_shaders(directory: &Path) -> ReShaderResult<()> {
+    let merge_directory = directory.join("Merged");
+    if !merge_directory.exists() {
+        std::fs::create_dir(&merge_directory)?;
+    }
+    let shader_directory = merge_directory.join("Shaders");
+    if !shader_directory.exists() {
+        std::fs::create_dir(&shader_directory)?;
+    }
+    let texture_directory = merge_directory.join("Textures");
+    if !texture_directory.exists() {
+        std::fs::create_dir(&texture_directory)?;
+    }
+    let intermediate_directory = merge_directory.join("Intermediate");
+    if !intermediate_directory.exists() {
+        std::fs::create_dir(&intermediate_directory)?;
+    }
+
+    for shader in SHADER_REPOSITORIES.iter() {
+        shader.clone_repo(directory)?;
+        shader.pull(directory)?;
+
+        let repo_directory = directory.join(&shader.name);
+        let repo_shader_directory = repo_directory.join("Shaders");
+        let repo_texture_directory = repo_directory.join("Textures");
+        let target_directory = if shader.essential {
+            shader_directory.clone()
+        } else {
+            shader_directory.join(&shader.name)
+        };
+        if !target_directory.exists() {
+            std::fs::create_dir(&target_directory)?;
+        }
+
+        if repo_shader_directory.exists() {
+            let builder = CopyBuilder::new(&repo_shader_directory, &target_directory);
+            builder.overwrite(true).run()?;
+        }
+
+        if repo_texture_directory.exists() {
+            let builder = CopyBuilder::new(&repo_texture_directory, &texture_directory);
+            builder.overwrite(true).run()?;
```

Comment:
We run into an ordering issue here between `crosire/reshade-shaders` and `CeeJayDK/SweetFX`, where if one repo had overlapping files that were not exactly the same then the order in which we add them matters. It stands to reason that it's likely for `crosire/reshade-shaders` to be the base directory over which things get overwritten, if needed. Currently I believe the ordering of repos in `SHADER_REPOSITORIES` makes it so `SweetFX` would get overwritten.

And indeed, if you run the following (assuming all pulled repositories are in the CWD at start)....
```
$ cd SweetFX
$ shasum Shaders/* > ../SweetFX-Shaders-SHAsums.txt
$ cd ../reshade-shaders
$ shasum -qc ../SweetFX-Shaders-SHAsums.txt --ignore-missing
Shaders/FXAA.fx: FAILED
Shaders/Nostalgia.fx: FAILED
Shaders/SMAA.fx: FAILED
shasum: WARNING: 3 computed checksums did NOT match
```
So at time of writing three files in `SweetFX` that are also in `crosire/reshade-shaders` on the `master` branch do *not* match.

I checked the SHAs versus a copy of `reshade-shaders` from a fresh install on Windows that a friend provided me and `SweetFX` shaders match perfectly, whereas `crosire/reshade-shaders` did not match for 6 files (the 3 noted above that conflict between SweetFX and itself, as well as 3 additional files that are newer in the fresh install than what appears to be currently on `master` in `crosire/reshade-shaders`. These files appear to be from the `slim` branch.).

For what it's worth, there is _no_ overlap currently between `SweetFX` and the `slim` branch, and it seems none of the files on `master` that are not in `slim` are in the fresh install. So I think this may be a case of the infrastructure having changed since the time of the shell script my original suggestion was based off of. Indeed that script actually didn't even make the subdirectories currently seen, it just shoved everything into one flat `Shaders` directory, though it did not overwrite anything.

Tl;dr it seems we need to use the `slim` branch to conform most closely to what a pristine installation does, and regardless we want `crosire/reshade-shaders` to go first / get overwritten when a conflict emerges. If we did want to ship the extra shaders `master` has over `slim` it would require some finangling since `slim` is definitely newer than `master` is currently.

Also having said all that... it seems this file is probably our canonical information sheet on what to install and how: https://github.com/crosire/reshade-shaders/blob/list/EffectPackages.ini

From the `list` branch of `crosire/reshade-shaders`.

So... that's an oops on my part sorry.